src/script/conversation/ConversationRepository.ts

```diff
@@ -1742,9 +1742,13 @@ export class ConversationRepository {
    */
   public async leaveConversation(conversation: Conversation, {localOnly = false} = {}) {
     const userQualifiedId = this.userState.self().qualifiedId;
-    return localOnly
-      ? this.removeMembersLocally(conversation, [userQualifiedId])
-      : this.removeMembers(conversation, [userQualifiedId]);
+
+    if (localOnly) {
+      return this.removeMembersLocally(conversation, [userQualifiedId]);
+    }
+
+    const events = await this.removeMembersFromConversation(conversation, [userQualifiedId]);
```

Comment:
We cannot call `.removeMembers` method. We want to always delete the user "the proteus way", no need to check for conversation protocol.